### ISSUE TITLE
Netty memory leak with server sent events

### ISSUE THREAD

**Issue by tchiotludo** [micronaut-projects/micronaut-core#7098]
We have an error that is raised several time per day on several project : 
`LEAK: ByteBuf.release() was not called before it's garbage-collected`: 

The project are : 
- https://github.com/tchiotludo/akhq with [this issue](https://github.com/tchiotludo/akhq/issues/1037)
- https://github.com/kestra-io/kestra

I can't make it easily reproduce but I'm sure that this error is located on Server Sent Event.

The stacktrace : 
```
LEAK: ByteBuf.release() was not called before it's garbage-collected. See https://netty.io/wiki/reference-counted-objects.html for more information.
Recent access records: 
Created at:
  io.netty.buffer.PooledByteBufAllocator.newDirectBuffer(PooledByteBufAllocator.java:401)
  io.netty.buffer.AbstractByteBufAllocator.directBuffer(AbstractByteBufAllocator.java:188)
  io.netty.buffer.AbstractByteBufAllocator.directBuffer(AbstractByteBufAllocator.java:174)
  io.netty.buffer.AbstractByteBufAllocator.buffer(AbstractByteBufAllocator.java:108)
  io.micronaut.buffer.netty.NettyByteBufferFactory.buffer(NettyByteBufferFactory.java:77)
  io.micronaut.json.codec.MapperMediaTypeCodec.encode(MapperMediaTypeCodec.java:255)
  io.micronaut.http.server.codec.TextStreamCodec.encode(TextStreamCodec.java:158)
  io.micronaut.http.codec.MediaTypeCodec.encode(MediaTypeCodec.java:128)
  io.micronaut.http.server.netty.RoutingInBoundHandler.lambda$mapToHttpContent$7(RoutingInBoundHandler.java:1046)
  reactor.core.publisher.FluxMapFuseable$MapFuseableSubscriber.onNext(FluxMapFuseable.java:113)
  io.micronaut.reactive.reactor.instrument.ReactorSubscriber.onNext(ReactorSubscriber.java:57)
  reactor.core.publisher.FluxHide$SuppressFuseableSubscriber.onNext(FluxHide.java:137)
  reactor.core.publisher.FluxPublishOn$PublishOnSubscriber.runAsync(FluxPublishOn.java:440)
  reactor.core.publisher.FluxPublishOn$PublishOnSubscriber.run(FluxPublishOn.java:527)
  io.micronaut.reactive.reactor.instrument.ReactorInstrumentation.lambda$null$0(ReactorInstrumentation.java:62)
  reactor.core.scheduler.WorkerTask.call(WorkerTask.java:84)
  reactor.core.scheduler.WorkerTask.call(WorkerTask.java:37)
  io.micronaut.scheduling.instrument.InvocationInstrumenterWrappedCallable.call(InvocationInstrumenterWrappedCallable.java:53)
  io.micrometer.core.instrument.composite.CompositeTimer.recordCallable(CompositeTimer.java:68)
  io.micrometer.core.instrument.Timer.lambda$wrap$1(Timer.java:171)
  java.base/java.util.concurrent.FutureTask.run(Unknown Source)
  java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
  java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
  java.base/java.lang.Thread.run(Unknown Source)
```

Do you need more information or only the stacktrace will help you ?

### Version

3.3.4

**Comment by graemerocher** [comment]
@yawkat can you take a look next time you have cycles? Thanks

**Comment by yawkat** [comment]
Another that is likely fixed by #6896
